src/Features/Core/Portable/PdbSourceDocument/DocumentDebugInfoReader.cs

```diff
@@ -47,13 +48,44 @@ public ImmutableArray<SourceDocument> FindSourceDocuments(ISymbol symbol)
                 var checksum = _pdbReader.GetBlobContent(document.Hash);
 
                 var embeddedTextBytes = TryGetEmbeddedTextBytes(handle);
+                var sourceLinkUrl = TryGetSourceLinkUrl(handle);
 
-                sourceDocuments.Add(new SourceDocument(filePath, hashAlgorithm, checksum, embeddedTextBytes));
+                sourceDocuments.Add(new SourceDocument(filePath, hashAlgorithm, checksum, embeddedTextBytes, sourceLinkUrl));
             }
 
             return sourceDocuments.ToImmutable();
         }
 
+        private string? TryGetSourceLinkUrl(DocumentHandle handle)
+        {
+            var document = _pdbReader.GetDocument(handle);
+            if (document.Name.IsNil)
+                return null;
+
+            var documentName = _pdbReader.GetString(document.Name);
```

Comment:
It's been a bit since I looked at the contract, but does _pdbReader throw if you provide a name that `isNil`?

Reply:
I don't know, but I'm checking it just above. Or were you suggesting the check is redundant? I'm a belt and braces kinda person I guess (except when it comes to network and file IO which I seem to always forget about)

Reply:
This was just a curiosity. I think one of the checks is not needed but I'm not sure enough to tell you to change it